include/boost/spirit/home/lex/lexer/lexertl/functor_data.hpp

```diff
@@ -68,7 +68,7 @@ namespace boost { namespace spirit { namespace lex { namespace lexertl
             {
 // some (random) versions of gcc instantiate this function even if it's not 
 // needed leading to false static asserts
-#if !defined(__GNUC__)
+#if !defined(__GNUC__) && !defined(BOOST_EMBTC)
```

Comment:
I do not understand why this is needed, both `clang++` and `clang-cl` does not fire an assertion in
```cpp
template <typename T>
void foo(T)
{
  static_assert(false, "");
}
```
However the code itself is ill-formed and needs a rewrite...

Reply:
I was just being careful because clang on Linux and clang on Windows targeting gcc both define \_\_GNUC\_\_ and the Embarcadero C++ clang-based compilers define \_\_clang\_\_ but do not define \_\_GNUC\_\_. However I can remove this case as long as I know it does not also refer to clang. I hope you do realize that \_\_GNUC\_\_ does refer to clang also in the vast majority, unless you separately check that \_\_clang\_\_ is or is not also defined also.

Reply:
I picked up the latest 'develop' changes and merged them in my PR.